### PR TITLE
Resolve #2960

### LINKED ISSUE #2960: K.std results in NaN if input is zero vector
Since `K.std(x)` is just `sqrt(var(x))` under the hood the gradient will be `inf` if the input vector `x` is 0.

This is problematic when using the `BatchNormalization` layer and having a model that sometimes produces zero vectors.

Wouldn't it be better to expose `K.var` in the backends and use that instead of `K.std` so that
the code can do `K.sqrt(K.var(x) + eps)` and does not have to add `eps` to `x`?

### PR DESCRIPTION
**Pull request by githubnemo**
Introduce `K.var` so that the standard deviation computation can
be made numerically stable. Instead of

	K.std(x)

the user is able to write

	K.sqrt(K.var(x) + self.epsilon)

avoiding a division by zero in the gradient computation of `sqrt`.

### CODE REVIEW

File: keras/backend/tensorflow_backend.py [hunk 1]
```diff
@@ -314,17 +314,23 @@ def prod(x, axis=None, keepdims=False):
     return tf.reduce_prod(x, reduction_indices=axis, keep_dims=keepdims)
 
 
-def std(x, axis=None, keepdims=False):
-    '''Standard deviation of a tensor, alongside the specificied axis.
+def var(x, axis=None, keepdims=False):
+    '''Variance of a tensor, alongside the specificied axis.
```
Comment (fchollet) [inline keras/backend/tensorflow_backend.py]: "specified"

**Comment by fchollet** [comment]
LGTM

File: keras/layers/normalization.py [hunk 2]
```diff
@@ -139,7 +139,7 @@ def call(self, x, mask=None):
         elif self.mode == 1:
             # sample-wise normalization
             m = K.mean(x, axis=-1, keepdims=True)
-            std = K.std(x, axis=-1, keepdims=True)
+            std = K.sqrt(K.var(x, axis=-1, keepdims=True) + self.epsilon)
             x_normed = (x - m) / (std + self.epsilon)
```
Comment (gag) [inline keras/layers/normalization.py]: Is that OK that self.epsilon is added twice: once while calculating std thus preventing is from being zero and then later when normalizing x?
